### PR TITLE
Improve logic around applying defaults

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #none# - documentation fixes and/or test additions
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
* **Improved** application of default values when processing `.pr-bumper.json`. Whereas previously you would have to provide an entire, complex, nested object if you wanted to override the default for any part of a config object, you can now override a partial object in your `.pr-bumper.json` and the rest of the object will get the appropriate defaults.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -36,20 +56,30 @@ function getChangelogSectionIndex (lines) {
 }
 
 /**
+ * Get the given key from process.env, substituting "undefined" for the real undefined
+ * @param {String} key - the environment variable we want to get
+ * @returns {*} whatever is at process.env[key] with the one exception of "undefined" being translated to undefined
+ */
+function getEnv (key) {
+  const value = process.env[key]
+  return (value === 'undefined') ? undefined : value
```
Comment (sandersky) [inline lib/utils.js]: I see now you check against the string `undefined` whereas before we used `||`, checking against the value actually being undefined/null/0. Does this mean defaults never actually worked before?

Comment (job13er) [inline lib/utils.js]: I think they did, but in some cases we got an "undefined" string when really we wanted `undefined`. In this case, an undefined/null/0 will still get through, but a string "undefined" will also be considered `undefined`. I noticed that in some cases, the `process.env` ended up with "undefined" values when I didn't think they should be set at all.

**Comment by sandersky** [comment]
:+1:

**Comment by job13er** [comment]
Approved

**Pull request merged**